src/Model/Notify.php

```diff
@@ -0,0 +1,775 @@
+			'network', 'created', 'object', 'parent-author-name', 'parent-author-link', 'parent-guid'];
+		$params = ['order' => ['received' => true], 'limit' => [$start, $limit]];
+
+		$items = Item::selectForUser(local_user(), $fields, $condition, $params);
+
+		if ($this->dba->isResult($items)) {
+			$notifies = $this->formatNotifies(Item::inArray($items), $ident);
+		}
+
+		$arr = [
+			'notifications' => $notifies,
+			'ident'         => $ident,
+		];
+
+		return $arr;
+	}
+
+	/**
+	 * @brief Get introductions
+	 *
+	 * @param bool $all     If false only include introductions into the query
+	 *                      which aren't marked as ignored
+	 * @param int  $start   Start the query at this point
+	 * @param int  $limit   Maximum number of query results
+	 * @param int  $id      When set, only the introduction with this id is displayed
+	 *
+	 * @return array [string, array]
+	 *    string 'ident' => Notification identifier
+	 *    array 'notifications' => Introductions
+	 *
+	 * @throws ImagickException
+	 * @throws Exception
+	 */
+	public function getIntroNotifies(bool $all = false, int $start = 0, int $limit = self::DEFAULT_PAGE_LIMIT, int $id = 0)
+	{
+		/// @todo sanitize wording according to SELF::INTRO
+		$ident     = 'introductions';
+		$notifies  = [];
+		$sql_extra = "";
+
+		if (empty($id)) {
+			if (!$all) {
+				$sql_extra = " AND NOT `ignore` ";
+			}
+
+			$sql_extra .= " AND NOT `intro`.`blocked` ";
+		} else {
+			$sql_extra = sprintf(" AND `intro`.`id` = %d ", intval($id));
+		}
+
+		/// @todo Fetch contact details by "Contact::getDetailsByUrl" instead of queries to contact, fcontact and gcontact
+		$stmtNotifies = $this->dba->p(
+			"SELECT `intro`.`id` AS `intro_id`, `intro`.*, `contact`.*,
+				`fcontact`.`name` AS `fname`, `fcontact`.`url` AS `furl`, `fcontact`.`addr` AS `faddr`,
+				`fcontact`.`photo` AS `fphoto`, `fcontact`.`request` AS `frequest`,
+				`gcontact`.`location` AS `glocation`, `gcontact`.`about` AS `gabout`,
+				`gcontact`.`keywords` AS `gkeywords`, `gcontact`.`gender` AS `ggender`,
+				`gcontact`.`network` AS `gnetwork`, `gcontact`.`addr` AS `gaddr`
+			FROM `intro`
+				LEFT JOIN `contact` ON `contact`.`id` = `intro`.`contact-id`
+				LEFT JOIN `gcontact` ON `gcontact`.`nurl` = `contact`.`nurl`
+				LEFT JOIN `fcontact` ON `intro`.`fid` = `fcontact`.`id`
+			WHERE `intro`.`uid` = ? $sql_extra
+			LIMIT ?, ?",
+			$_SESSION['uid'],
+			$start,
+			$limit
+		);
+		if ($this->dba->isResult($stmtNotifies)) {
+			$notifies = $this->formatIntros($this->dba->toArray($stmtNotifies));
+		}
+
+		$arr = [
+			'ident'         => $ident,
+			'notifications' => $notifies,
+		];
+
+		return $arr;
+	}
+
+	/**
+	 * @brief Format the notification query in an usable array
+	 *
+	 * @param array $intros The array from the db query
+	 *
+	 * @return array with the introductions
+	 * @throws HTTPException\InternalServerErrorException
+	 * @throws ImagickException
+	 */
+	private function formatIntros(array $intros)
```

Comment:
Seeing the naming pattern I created with my earlier requests, this probably should be renamed `formatIntroList`.